doc/cookbook/source/examples/regression/least_squares_regression.rst

```diff
@@ -0,0 +1,33 @@
+========================
+Least Squares Regression
+========================
+
+A Linear regression model can be defined as :math:`y_i = \bf{w}.\bf{x_i}` where :math:`y_i` is the predicted value, :math:`\bf{x_i}` is a feature vector and :math:`\bf{w}` is the weight vector. We aim to find the linear function that best explains the data, i.e. minimizes the error or loss function :math:`E(\bf{w})` by finding appropriate :math:`\bf{w}`.
+
+.. math::
+    E({\bf{w}}) = \sum_{i=1}^N(y_i-{\bf w}\cdot {\bf x}_i)^2
+
+-------
+Example
+-------
+
+Imagine we have files with training and test data. We create `CDenseFeatures` (here 64 bit floats aka RealFeatures) and :sgclass:`CRegressionLabels` as
```

Comment:
Putting :sgclass:`CDenseFeatures` gives link error.  It's not there in `knn.sg` also.